[CELEBORN-1190][FOLLOWUP] Fix WARNING of error prone

SteNicholas opened the pull request:
### What changes were proposed in this pull request?

- Fix `WARNING` of error prone.
- Disable `EmptyCatch`, `JdkObsolete`, `MutableConstantField` and `UnnecessaryParentheses`. 

### Why are the changes needed?

There are many `WARNING` generated by error prone. We should follow the suggestion of error prone to fix `WARNING`.

```
$ mvn clean install -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/network/sasl/SaslUtils.java:[44,25] [MutableConstantField] Constant field declarations should use the immutable type (such as ImmutableList) instead of the general collection interface type (such as List)
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/network/sasl/SaslUtils.java:[47,18] [MutableConstantField] Constant field declarations should use the immutable type (such as ImmutableList) instead of the general collection interface type (such as List)
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/network/client/TransportClientBootstrap.java:[34,5] [InvalidParam] Parameter name `channel` is unknown.
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java:[96,29] [StaticAssignmentInConstructor] This assignment is to a static field. Mutating static state from a constructor is highly error-prone.
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java:[104,30] [StaticAssignmentInConstructor] This assignment is to a static field. Mutating static state from a constructor is highly error-prone.
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/network/sasl/anonymous/AnonymousSaslServerFactory.java:[67,2] [ClassCanBeStatic] Inner class is non-static but does not reference enclosing class
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java:[60,17] [NonAtomicVolatileUpdate] This update of a volatile variable is non-atomic
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/network/util/TransportFrameDecoder.java:[54,46] [JdkObsolete] It is very rare for LinkedList to out-perform ArrayList or ArrayDeque. Avoid it unless you're willing to invest a lot of time into benchmarking. Caveat: LinkedList supports null elements, but ArrayDeque does not.
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/network/ssl/ReloadingX509TrustManager.java:[207,29] [NonAtomicVolatileUpdate] This update of a volatile variable is non-atomic
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/network/ssl/ReloadingX509TrustManager.java:[216,28] [NonAtomicVolatileUpdate] This update of a volatile variable is non-atomic
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/network/sasl/anonymous/AnonymousSaslClientFactory.java:[73,2] [ClassCanBeStatic] Inner class is non-static but does not reference enclosing class
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/network/sasl/anonymous/AnonymousSaslClientFactory.java:[93,31] [DefaultCharset] Implicit use of the platform default charset, which can result in differing behaviour between JVM executions or incorrect behavior if the encoding of the data source doesn't match expectations.
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/util/ExceptionUtils.java:[65,11] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/common/src/main/java/org/apache/celeborn/common/util/ExceptionUtils.java:[66,11] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/common/src/test/java/org/apache/celeborn/common/network/ssl/SslSampleConfigs.java:[164,16] [JavaUtilDate] Date has a bad API that leads to bugs; prefer java.time.Instant or LocalDate.
[WARNING] /Users/nicholas/Github/celeborn/common/src/test/java/org/apache/celeborn/common/network/ssl/SslSampleConfigs.java:[165,14] [JavaUtilDate] Date has a bad API that leads to bugs; prefer java.time.Instant or LocalDate.
[WARNING] /Users/nicholas/Github/celeborn/common/src/test/java/org/apache/celeborn/common/network/ssl/SslSampleConfigs.java:[165,35] [JavaUtilDate] Date has a bad API that leads to bugs; prefer java.time.Instant or LocalDate.
[WARNING] /Users/nicholas/Github/celeborn/common/src/test/java/org/apache/celeborn/common/network/SSLTransportClientFactorySuiteJ.java:[32,14] [MissingOverride] setUp overrides method in TransportClientFactorySuiteJ; expected @Override
[WARNING] /Users/nicholas/Github/celeborn/common/src/test/java/org/apache/celeborn/common/network/SSLTransportClientFactorySuiteJ.java:[40,14] [MissingOverride] tearDown overrides method in TransportClientFactorySuiteJ; expected @Override
[WARNING] /Users/nicholas/Github/celeborn/common/src/test/java/org/apache/celeborn/common/network/protocol/EncryptedMessageWithHeaderSuiteJ.java:[124,6] [UseCorrectAssertInTests] Java assert is used in test. For testing purposes Assert.* matchers should be used.
[WARNING] /Users/nicholas/Github/celeborn/common/src/test/java/org/apache/celeborn/common/network/RpcIntegrationSuiteJ.java:[255,15] [UnusedMethod] Private method 'assertErrorAndClosed' is never used.
[WARNING] /Users/nicholas/Github/celeborn/common/src/test/java/org/apache/celeborn/common/network/RpcIntegrationSuiteJ.java:[154,17] [UnusedNestedClass] This nested class is unused, and can be removed.
[WARNING] /Users/nicholas/Github/celeborn/common/src/test/java/org/apache/celeborn/common/network/RpcIntegrationSuiteJ.java:[57,15] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/common/src/test/java/org/apache/celeborn/common/network/ssl/ReloadingX509TrustManagerSuiteJ.java:[107,10] [AssertThrowsMultipleStatements] The lambda passed to assertThrows should contain exactly one statement
[WARNING] /Users/nicholas/Github/celeborn/common/src/test/java/org/apache/celeborn/common/network/ssl/ReloadingX509TrustManagerSuiteJ.java:[134,10] [AssertThrowsMultipleStatements] The lambda passed to assertThrows should contain exactly one statement
[WARNING] /Users/nicholas/Github/celeborn/client/src/main/java/org/apache/celeborn/client/read/LocalPartitionReader.java:[84,31] [StaticAssignmentInConstructor] This assignment is to a static field. Mutating static state from a constructor is highly error-prone.
[WARNING] /Users/nicholas/Github/celeborn/client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:[130,6] [ThreadLocalUsage] ThreadLocals should be stored in static fields
[WARNING] /Users/nicholas/Github/celeborn/client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:[714,6] [MissingCasesInEnumSwitch] Non-exhaustive switch; either add a default or handle the remaining cases: SUCCESS, PARTIAL_SUCCESS, REQUEST_FAILED, and 43 others
[WARNING] /Users/nicholas/Github/celeborn/client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:[1609,10] [MissingCasesInEnumSwitch] Non-exhaustive switch; either add a default or handle the remaining cases: PARTIAL_SUCCESS, REQUEST_FAILED, SHUFFLE_ALREADY_REGISTERED, and 45 others
[WARNING] /Users/nicholas/Github/celeborn/client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:[1648,26] [MissingOverride] updateFileGroup implements method in ShuffleClient; expected @Override
[WARNING] /Users/nicholas/Github/celeborn/client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:[1654,57] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:[1823,32] [MissingOverride] getDataClientFactory implements method in ShuffleClient; expected @Override
[WARNING] /Users/nicholas/Github/celeborn/client/src/test/java/org/apache/celeborn/client/ShuffleClientSuiteJ.java:[185,6] [UseCorrectAssertInTests] Java assert is used in test. For testing purposes Assert.* matchers should be used.
[WARNING] /Users/nicholas/Github/celeborn/service/src/main/java/org/apache/celeborn/server/common/service/store/db/DbServiceManagerImpl.java:[70,33] [JavaUtilDate] Date has a bad API that leads to bugs; prefer java.time.Instant or LocalDate.
[WARNING] /Users/nicholas/Github/celeborn/service/src/main/java/org/apache/celeborn/server/common/service/store/db/DbServiceManagerImpl.java:[71,33] [JavaUtilDate] Date has a bad API that leads to bugs; prefer java.time.Instant or LocalDate.
[WARNING] /Users/nicholas/Github/celeborn/master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/HARaftServer.java:[424,11] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/HARaftServer.java:[425,11] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/HARaftServer.java:[496,55] [UnescapedEntity] This looks like a type with type parameters. The < and > characters here will be interpreted as HTML, which can be avoided by wrapping it in a {@code } tag.
[WARNING] /Users/nicholas/Github/celeborn/master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/SingleMasterMetaManager.java:[166,14] [MissingOverride] handleUpdatePartitionSize implements method in IMetadataHandler; expected @Override
[WARNING] /Users/nicholas/Github/celeborn/master/src/main/java/org/apache/celeborn/service/deploy/master/SlotsAllocator.java:[298,61] [JdkObsolete] It is very rare for LinkedList to out-perform ArrayList or ArrayDeque. Avoid it unless you're willing to invest a lot of time into benchmarking. Caveat: LinkedList supports null elements, but ArrayDeque does not.
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/MapPartitionDataReader.java:[346,37] [NonAtomicVolatileUpdate] This update of a volatile variable is non-atomic
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java:[202,33] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java:[300,31] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java:[497,17] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java:[503,17] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/memory/MemoryManager.java:[513,39] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/CreditStreamManager.java:[256,12] [ClassCanBeStatic] Inner class is non-static but does not reference enclosing class
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/WorkerSecretRegistryImpl.java:[73,12] [CacheLoaderNull] The result of CacheLoader#load must be non-null.
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/ReducePartitionDataWriter.java:[69,13] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/ReducePartitionDataWriter.java:[73,13] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/ReducePartitionDataWriter.java:[103,24] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/ReducePartitionDataWriter.java:[104,39] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/MapPartitionDataWriter.java:[261,46] [ByteBufferBackingArray] ByteBuffer.array() shouldn't be called unless ByteBuffer.arrayOffset() is used or if the ByteBuffer was initialized using ByteBuffer.wrap() or ByteBuffer.allocate().
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/ChunkStreamManager.java:[102,40] [NonAtomicVolatileUpdate] This update of a volatile variable is non-atomic
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/ChunkStreamManager.java:[109,40] [NonAtomicVolatileUpdate] This update of a volatile variable is non-atomic
[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java:[318,39] [IntLongMath] Expression of type int may overflow before being assigned to a long
[WARNING] /Users/nicholas/Github/celeborn/worker/src/test/java/org/apache/celeborn/service/deploy/worker/FetchHandlerSuiteJ.java:[133,6] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/worker/src/test/java/org/apache/celeborn/service/deploy/worker/network/SSLRequestTimeoutIntegrationSuiteJ.java:[32,14] [MissingOverride] setUp overrides method in RequestTimeoutIntegrationSuiteJ; expected @Override
[WARNING] /Users/nicholas/Github/celeborn/worker/src/test/java/org/apache/celeborn/service/deploy/worker/network/SSLRequestTimeoutIntegrationSuiteJ.java:[40,14] [MissingOverride] tearDown overrides method in RequestTimeoutIntegrationSuiteJ; expected @Override
[WARNING] /Users/nicholas/Github/celeborn/worker/src/test/java/org/apache/celeborn/service/deploy/worker/storage/ChunkFetchIntegrationSuiteJ.java:[74,15] [UnnecessaryParentheses] These grouping parentheses are unnecessary; it is unlikely the code will be misinterpreted without them
[WARNING] /Users/nicholas/Github/celeborn/worker/src/test/java/org/apache/celeborn/service/deploy/worker/storage/ChunkFetchIntegrationSuiteJ.java:[186,47] [JdkObsolete] It is very rare for LinkedList to out-perform ArrayList or ArrayDeque. Avoid it unless you're willing to invest a lot of time into benchmarking. Caveat: LinkedList supports null elements, but ArrayDeque does not.
[WARNING] /Users/nicholas/Github/celeborn/worker/src/test/java/org/apache/celeborn/service/deploy/worker/storage/SSLReducePartitionDataWriterSuiteJ.java:[30,26] [MissingOverride] createModuleTransportConf overrides method in DiskReducePartitionDataWriterSuiteJ; expected @Override
[WARNING] /Users/nicholas/Github/celeborn/worker/src/test/java/org/apache/celeborn/service/deploy/worker/storage/local/DiskReducePartitionDataWriterSuiteJ.java:[234,47] [JdkObsolete] It is very rare for LinkedList to out-perform ArrayList or ArrayDeque. Avoid it unless you're willing to invest a lot of time into benchmarking. Caveat: LinkedList supports null elements, but ArrayDeque does not.
[WARNING] /Users/nicholas/Github/celeborn/worker/src/test/java/org/apache/celeborn/service/deploy/worker/storage/memory/MemoryReducePartitionDataWriterSuiteJ.java:[198,47] [JdkObsolete] It is very rare for LinkedList to out-perform ArrayList or ArrayDeque. Avoid it unless you're willing to invest a lot of time into benchmarking. Caveat: LinkedList supports null elements, but ArrayDeque does not.
```
```
$ mvn clean install -Pspark-2.4 -pl client-spark/common,client-spark/spark-2 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
[WARNING] /Users/nicholas/Github/celeborn/client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedPusher.java:[109,57] [JdkObsolete] It is very rare for LinkedList to out-perform ArrayList or ArrayDeque. Avoid it unless you're willing to invest a lot of time into benchmarking. Caveat: LinkedList supports null elements, but ArrayDeque does not.
[WARNING] /Users/nicholas/Github/celeborn/client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SendBufferPool.java:[56,14] [JdkObsolete] It is very rare for LinkedList to out-perform ArrayList or ArrayDeque. Avoid it unless you're willing to invest a lot of time into benchmarking. Caveat: LinkedList supports null elements, but ArrayDeque does not.
[WARNING] /Users/nicholas/Github/celeborn/client-spark/common/src/main/java/org/apache/spark/shuffle/celeborn/SendBufferPool.java:[57,21] [JdkObsolete] It is very rare for LinkedList to out-perform ArrayList or ArrayDeque. Avoid it unless you're willing to invest a lot of time into benchmarking. Caveat: LinkedList supports null elements, but ArrayDeque does not.
[WARNING] /Users/nicholas/Github/celeborn/client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java:[247,14] [UnusedMethod] Private method 'executorCores' is never used.
[WARNING] /Users/nicholas/Github/celeborn/client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java:[120,55] [ReferenceEquality] Comparison using reference equality instead of value equality
```
```
$ mvn clean install -Pspark-3.5 -pl client-spark/spark-3 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
[WARNING] /Users/nicholas/Github/celeborn/client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/CelebornShuffleDataIO.java:[65,17] [MissingOverride] supportsReliableStorage implements method in ShuffleDriverComponents; expected @Override
[WARNING] /Users/nicholas/Github/celeborn/client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java:[163,55] [ReferenceEquality] Comparison using reference equality instead of value equality
```
```
$ mvn clean install -Pflink-1.14 -pl client-flink/common,client-flink/flink-1.14 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/readclient/CelebornBufferStream.java:[161,17] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/readclient/CelebornBufferStream.java:[223,27] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/ShuffleTaskInfo.java:[46,17] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleInputGateDelegation.java:[99,66] [JdkObsolete] It is very rare for LinkedList to out-perform ArrayList or ArrayDeque. Avoid it unless you're willing to invest a lot of time into benchmarking. Caveat: LinkedList supports null elements, but ArrayDeque does not.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleInputGateDelegation.java:[236,21] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleInputGateDelegation.java:[251,19] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleInputGateDelegation.java:[267,17] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleInputGateDelegation.java:[354,17] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleInputGateDelegation.java:[392,17] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleInputGateDelegation.java:[473,17] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleInputGateDelegation.java:[533,17] [SynchronizeOnNonFinalField] Synchronizing on non-final fields is not safe: if the field is ever updated, different threads may end up locking on different objects.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/buffer/TransferBufferPool.java:[182,33] [MixedMutabilityReturnType] This method returns both mutable and immutable collections or maps from different paths. This may be confusing for users of the method.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/main/java/org/apache/celeborn/plugin/flink/utils/FlinkUtils.java:[34,6] [DoubleBraceInitialization] Prefer collection factory methods or builders to the double-brace initialization pattern.
[WARNING] /Users/nicholas/Github/celeborn/client-flink/common/src/test/java/org/apache/celeborn/plugin/flink/BufferPackSuiteJ.java:[207,6] [CatchAndPrintStackTrace] Logging or rethrowing exceptions should usually be preferred to catching and calling printStackTrace
[WARNING] /Users/nicholas/Github/celeborn/client-flink/flink-1.14/src/test/java/org/apache/celeborn/plugin/flink/RemoteShuffleResultPartitionSuiteJ.java:[140,67] [CanonicalDuration] Duration can be expressed more clearly with different units
```
```
$ mvn clean install -Pflink-1.15 -pl client-flink/flink-1.15 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
[WARNING] /Users/nicholas/Github/celeborn/client-flink/flink-1.15/src/test/java/org/apache/celeborn/plugin/flink/RemoteShuffleResultPartitionSuiteJ.java:[140,67] [CanonicalDuration] Duration can be expressed more clearly with different units
```
```
$ mvn clean install -Pflink-1.17 -pl client-flink/flink-1.16 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
```
```
$ mvn clean install -Pflink-1.17 -pl client-flink/flink-1.17 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
[WARNING] /Users/nicholas/Github/celeborn/client-flink/flink-1.17/src/test/java/org/apache/celeborn/plugin/flink/RemoteShuffleResultPartitionSuiteJ.java:[140,67] [CanonicalDuration] Duration can be expressed more clearly with different units
```
```
$ mvn clean install -Pflink-1.18 -pl client-flink/flink-1.18 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
[WARNING] /Users/nicholas/Github/celeborn/client-flink/flink-1.18/src/test/java/org/apache/celeborn/plugin/flink/RemoteShuffleResultPartitionSuiteJ.java:[140,67] [CanonicalDuration] Duration can be expressed more clearly with different units
```
```
$ mvn clean install -Pflink-1.19 -pl client-flink/flink-1.19 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
[WARNING] /Users/nicholas/Github/celeborn/client-flink/flink-1.19/src/test/java/org/apache/celeborn/plugin/flink/RemoteShuffleResultPartitionSuiteJ.java:[140,67] [CanonicalDuration] Duration can be expressed more clearly with different units
```
```
$ mvn clean install -Pmr -pl client-mr/mr -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
```

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual test.

```
$ mvn clean install -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
$ mvn clean install -Pspark-2.4 -pl client-spark/common,client-spark/spark-2 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
$ mvn clean install -Pspark-3.5 -pl client-spark/spark-3 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
$ mvn clean install -Pflink-1.14 -pl client-flink/common,client-flink/flink-1.14 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
$ mvn clean install -Pflink-1.15 -pl client-flink/flink-1.15 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
$ mvn clean install -Pflink-1.16 -pl client-flink/flink-1.15 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
$ mvn clean install -Pflink-1.17 -pl client-flink/flink-1.17 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
$ mvn clean install -Pflink-1.18 -pl client-flink/flink-1.18 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
$ mvn clean install -Pflink-1.19 -pl client-flink/flink-1.19 -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
$ mvn clean install -Pmr -pl client-mr/mr -DskipTests -Dcheckstyle.skip=true -Drat.skip=true -Dspotless.check.skip=true|grep WARNING|grep java
```

SteNicholas commented:
Ping @AngersZhuuuu, @pan3793, @cxzl25, @mridulm, @cxzl25.

SteNicholas commented on client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java:
```diff
@@ -244,19 +242,6 @@ public <K, C> ShuffleReader<K, C> getReader(
     return _sortShuffleManager.getReader(handle, startPartition, endPartition, context);
   }
 
-  private int executorCores(SparkConf conf) {
```
This change is blocked by #2562. cc @cxzl25.

SteNicholas commented on client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java:
```diff
@@ -69,7 +69,7 @@ public SparkShuffleManager(SparkConf conf, boolean isDriver) {
     this.isDriver = isDriver;
     this.celebornConf = SparkUtils.fromSparkConf(conf);
-    this.cores = conf.getInt(SparkLauncher.EXECUTOR_CORES, 1);
+    this.cores = executorCores(conf);
```
This change is blocked by https://github.com/apache/celeborn/pull/2562. cc @cxzl25.

mridulm commented on client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:
```diff
@@ -126,13 +125,7 @@ public class ShuffleClientImpl extends ShuffleClient {
-        }
-      };
+  private ThreadLocal<Compressor> compressorThreadLocal;
 
```
The change in this class does not look related to error-prone.
Do we want to split it out ?

SteNicholas replied:
@mridulm, this change follows [ThreadLocalUsage](https://errorprone.info/bugpattern/ThreadLocalUsage).

mridulm commented on client/src/main/java/org/apache/celeborn/client/read/LocalPartitionReader.java:
```diff
@@ -51,7 +51,7 @@
 
   private static final Logger logger = LoggerFactory.getLogger(LocalPartitionReader.class);
-  private static volatile ThreadPoolExecutor readLocalShufflePool;
+  private volatile ThreadPoolExecutor readLocalShufflePool;
```
This is expected to be a static variable, not instance.

SteNicholas replied:
@mridulm, this static variable does not satisfy [StaticAssignmentInConstructor](https://errorprone.info/bugpattern/StaticAssignmentInConstructor).

mridulm replied:
That is a reasonable recommandation when sufficient care is not taken to address the issues.
Here, we are doing the right thing by making sure it is initialized in a safe manner the first time the intiial instance of the class is created

Also note that the current change is a regression - for each instance of `LocalPartitionReader` we will create a new thread pool.

mridulm commented on common/src/main/java/org/apache/celeborn/common/meta/FileInfo.java:
```diff
@@ -56,7 +56,7 @@ public long getFileLength() {
 
-  public void updateBytesFlushed(long bytes) {
+  public synchronized void updateBytesFlushed(long bytes) {
     bytesFlushed += bytes;
```
I am not sure about this change - it is not clear to me why we are locking on `this` here.
I will revisit this later if you believe this is the right change - not sure if subclasses of this abstract class need this ... if true, n general I would expect those kind of patterns to be difficult to evolve.

SteNicholas replied:
@mridulm, you could take a look at the description of this pull request, which has listed all WARNINGs.

mridulm replied:
Scratch that - this change makes sense: not sure what I missed when I reviewed it first time around :-(

mridulm commented on common/src/main/java/org/apache/celeborn/common/network/client/TransportResponseHandler.java:
```diff
@@ -63,11 +63,11 @@ public class TransportResponseHandler extends MessageHandler<ResponseMessage> {
   private final long fetchTimeoutCheckerInterval;
-  private static ScheduledExecutorService fetchTimeoutChecker = null;
+  private ScheduledExecutorService fetchTimeoutChecker;
   private ScheduledFuture fetchCheckerScheduleFuture;
```
Both of these are expected to be initialized once per class, not per instance

mridulm commented on common/src/main/java/org/apache/celeborn/common/network/sasl/anonymous/AnonymousSaslClientFactory.java:
```diff
@@ -90,7 +91,7 @@ public byte[] evaluateChallenge(byte[] challenge) throws SaslException {
       }
       isCompleted = true;
-      return ANONYMOUS.getBytes();
+      return ANONYMOUS.getBytes(StandardCharsets.UTF_8);
```
nice !

mridulm commented on common/src/main/java/org/apache/celeborn/common/network/ssl/ReloadingX509TrustManager.java:
```diff
@@ -200,20 +200,22 @@ public void run() {
-          } catch (Exception ex) {
-            logger.warn(
-                "Could not load truststore (keep using existing one) : " + ex.toString(), ex);
+      synchronized (this) {
```
Why do we need this `synchronized` ?

SteNicholas replied:
@mridulm, this change follows [NonAtomicVolatileUpdate](https://errorprone.info/bugpattern/NonAtomicVolatileUpdate).

mridulm replied:
This change is not required IMO. These `volatile` variables being updated only in this method.
Technically, they dont need to be `volatile` - but were made volatile only to ensure we can mutate them from tests, and they become visible.

Note that if the intent is to actually address the corner case race condition of mutation from the test (that is reasonable) - then we should wrap the mutations from `ReloadingX509TrustManagerSuiteJ` as well in `synchronized`

mridulm commented on common/src/test/java/org/apache/celeborn/common/network/sasl/SaslTestBase.java:
```diff
@@ -96,9 +96,7 @@ void authHelper(
-      if (error != null) {
-        throw error;
-      }
+      assertNull(error);
```
`assertNull` throws NPE - here we are throwing the `error` `Throwable`.
`testReRegisterationFails`, for example, requires the `error` to be thrown.

mridulm replied:
hmm, I am surprised the tests passed !
+CC @otterc

mridulm replied:
Oh wait, `Throwables.getStackTraceAsString` will serialize the 'cause' as well, heh ...
The intent here though is to throw the actual exception which was raised - not NPE :)

mridulm commented on common/src/test/java/org/apache/celeborn/common/network/ssl/ReloadingX509TrustManagerSuiteJ.java:
```diff
@@ -101,15 +101,7 @@ public void testLoadMissingTrustStore() throws Exception {
-            tm.destroy();
-          }
-        });
+        () -> new ReloadingX509TrustManager(KeyStore.getDefaultType(), trustStore, "password", 10));
```
This is not calling `init`, same below.

mridulm replied:
Ah, i see what you mean ... makes sense.

mridulm commented on worker/src/main/java/org/apache/celeborn/service/deploy/worker/WorkerSecretRegistryImpl.java:
```diff
@@ -65,12 +65,10 @@ public String load(String appId) {
-              // here.
               LOG.error(
                   "Failed to fetch the application meta info for {} from the master", appId, e);
+              throw new Exception(e);
```
This should be fine, but I will let Chandni comment better.

+CC @otterc

mridulm commented on worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/ChunkStreamManager.java:
```diff
@@ -96,14 +96,14 @@ public TimeWindow getFetchTimeMetric(long streamId) {
   }
 
-  public void chunkBeingSent(long streamId) {
+  public synchronized void chunkBeingSent(long streamId) {
     StreamState streamState = streams.get(streamId);
     if (streamState != null) {
       streamState.chunksBeingTransferred++;
     }
   }
```
This will be effectively a global lock, right ?
If the concern is around inconsistent update to `chunksBeingTransferred` (which is valid), why not lock `streamState` instead ?

```suggestion
  public void chunkBeingSent(long streamId) {
    StreamState streamState = streams.get(streamId);
    if (streamState != null) {
      synchronized(streamState) {
        streamState.chunksBeingTransferred++;
      }
    }
  }
```

(same below)

mridulm reviewed (commented):
Thanks for working on this @SteNicholas, this is super useful !
Had a query queries/comments.

venkata91 commented:
@SteNicholas Looks like this PR is marked stale without any activity. Are you still actively working on this one?

venkata91 commented on service/src/main/java/org/apache/celeborn/server/common/service/store/db/DbServiceManagerImpl.java:
```diff
@@ -61,6 +61,7 @@ public DbServiceManagerImpl(CelebornConf celebornConf, ConfigService configServe
     this.clusterId = createCluster(getClusterInfoFromEnv());
   }
 
+  @SuppressWarnings("JavaUtilDate")
```
Curious, why this is suppressed here? Is this planned to be addressed in a follow-up PR?

SteNicholas replied:
@venkata91, I referred to [JavaUtilDate](https://errorprone.info/bugpattern/JavaUtilDate) but this could not lead to bugs. Do you prefer the `java.time` APIs?

venkata91 replied:
Yes, migration to `java.time` APIs.

SteNicholas replied:
@mridulm, WDYT?

venkata91 replied:
Wouldn't it be better to move to `java.time` APIs to avoid any future evolution issues? Do you see any issues with migrating it now? Basically any backwards compatibility issues?

mridulm replied:
`Instant` and `LocalDate` is not directly handled by apache ibatis, is it ? I am not very confident - but given this is used in that context, we have to ensure it is supported (else we will end up writing type handler, etc ... I dont think it is worth that).

venkata91 replied:
@mridulm did you mean ratis or ibatis (looks like this is moved ro attic :))?

SteNicholas replied:
@mridulm, this change makes sense to you, right?

venkata91 replied:
@mridulm Looks like you meant `ibatis` only - seeing its usage in Celeborn.

venkata91 replied:
> Instant and LocalDate is not directly handled by apache ibatis, is it ? I am not very confident - but given this is used in that context, we have to ensure it is supported (else we will end up writing type handler, etc ... I dont think it is worth that).

I see the concern here. I'm fine with not handling this now given Apache

venkata91 replied:
@mridulm @SteNicholas Having said that, looks like the [type-handlers are merged in to mybatis core since 3.4.5](https://stackoverflow.com/questions/25113579/java-8-localdate-mapping-with-mybatis) [corresponding [PR](https://github.com/mybatis/mybatis-3/pull/974)] and Celeborn is already on 3.5.x - this looks to be fine, no?

SteNicholas replied:
@venkata91, changing to `Instant` type of [type-handlers are merged in to mybatis core since 3.4.5](https://stackoverflow.com/questions/25113579/java-8-localdate-mapping-with-mybatis) support has type compatibility problem for the database implementation of dynamic config. Right?

venkata91 replied:
Can you elaborate a bit? I don't think I i got it.

SteNicholas replied:
@venkata91, when `gmtCreate` and `gmtModify` are updated to `Instant` type. Does the `gmt_create` and `gmt_modify` in `celeborn-0.5.0-mysql.sql` need to change?

venkata91 replied:
Looks like `java.time.Instant` is mapped to TIMESTAMP based on the [InstantTypeHandlerTest](https://github.com/harawata/mybatis-3/blob/2da2275185e8bb185d51eb522120ac32e5635225/src/test/java/org/apache/ibatis/type/usesjava8/InstantTypeHandlerTest.java). Looks like it should be handled, no?

SteNicholas replied:
@venkata91, I have addressed comments in https://github.com/apache/celeborn/pull/2555/commits/9e0ff9fc000074ee6a78d0b3a91343125a89910a. `ConfigServiceSuiteJ` could verify this change. PTAL. cc @RexXiong.

venkata91 replied:
Looks good. Thanks!

venkata91 commented on worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/CreditStreamManager.java:
```diff
@@ -253,7 +253,7 @@ public int getActiveMapPartitionCount() {
   }
 
-  protected class StreamState {
+  protected static class StreamState {
```
Should this be `protected static` or just `static`?

SteNicholas replied:
@venkata91, this just follow `ClassCanBeStatic` rule. No need to remove protected.

SteNicholas commented:
@venkata91, I will address above comments today.

SteNicholas commented:
@mridulm, @venkata91, @FMX, I have rebased main branch and addressed above comments. PTAL.

SteNicholas commented:
@mridulm, @venkata91, @FMX, I have rebased main branch and addressed above comments in https://github.com/apache/celeborn/pull/2555/commits/d6acd18fca928bc02987aec79c989dc9fa4420c9. PTAL.

mridulm commented:
Thanks for the updates @SteNicholas ! Took another pass through it ... just a few comments outstanding from my end.

SteNicholas commented:
@mridulm, I have addressed your comments and pushed commit https://github.com/apache/celeborn/pull/2555/commits/e24964dbc0af12ba9b374397ac1f6ea2aa6067c1 with adding `@SuppressWarnings` annotation. PTAL.

venkata91 commented:
> @mridulm, I have addressed your comments and pushed commit [e24964d](https://github.com/apache/celeborn/commit/e24964dbc0af12ba9b374397ac1f6ea2aa6067c1) with adding `@SuppressWarnings` annotation. PTAL. @venkata91, any other comments for this pull request?

No other comments, @SteNicholas . Thanks for working on this!

SteNicholas commented on client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java:
```diff
@@ -126,6 +126,7 @@ public class ShuffleClientImpl extends ShuffleClient {
   private final TransportConf dataTransportConf;
 
+  @SuppressWarnings("ThreadLocalUsage")
   private final ThreadLocal<Compressor> compressorThreadLocal =
```
@mridulm, because `initialValue` depends on `conf`, `compressorThreadLocal` could not use `static`. Therefore this just use `@SuppressWarnings("ThreadLocalUsage")`.

SteNicholas commented:
@mridulm, could you please take a look at the changes in https://github.com/apache/celeborn/pull/2555/commits/950ec6ac1c5efdbb0cce1931c15f6d2b65f8cbb7. Thanks.

SteNicholas commented:
@venkata91, I have already used `Instant` instead of `Date`. PTAL.

venkata91 commented:
> @venkata91, I have already used `Instant` instead of `Date` in [749606f](https://github.com/apache/celeborn/commit/749606ff194224a26127920e13d96c3e46b72aed). PTAL.

lgtm. Thanks for addressing all the review comments.

SteNicholas commented:
> > @venkata91, I have already used `Instant` instead of `Date` in [749606f](https://github.com/apache/celeborn/commit/749606ff194224a26127920e13d96c3e46b72aed). PTAL.
> 
> lgtm. Thanks for addressing all the review comments.

Ping @mridulm.

mridulm reviewed (approved):
Looks good to me, thanks for fixing this @SteNicholas !

SteNicholas commented:
@FMX, @RexXiong, would you like to take a look?

mridulm commented:
If @FMX or @RexXiong are reviewing, will wait for them to be done - we can merge after.

FMX commented:
This PR is cool. I think we should add a new CI about error-prone tests to check all PRs to improve Celeborn's code quality.

RexXiong commented on service/src/main/java/org/apache/celeborn/server/common/service/model/ClusterInfo.java:
```diff
@@ -17,16 +17,16 @@
   private String endpoint;
-  private Date gmtCreate;
-  private Date gmtModify;
+  private Instant gmtCreate;
```
Are there compatibility issues when converting a timestamp to an Instant and vice versa using iBatis?

SteNicholas replied:
@RexXiong, I have added the verification in https://github.com/apache/celeborn/pull/2555/commits/6e9740c0663644e35c1b86fe7271792a7cdfd935. PTAL.

SteNicholas commented:
@FMX,  I will introduce CI error-prone test in another pull request. Meanwhile, I rebased the main branch and fix `[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/MapPartitionDataWriter.java:[58,21] [HidingField] Hiding fields of superclasses may cause confusion and errors. This field is hiding a field of the same name in superclass: PartitionDataWriter` in https://github.com/apache/celeborn/pull/2555/commits/6e9740c0663644e35c1b86fe7271792a7cdfd935. PTAL. cc @mridulm

SteNicholas commented:
@FMX,  I will introduce CI error-prone test in another pull request. Meanwhile, I rebased the main branch and fix `[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/MapPartitionDataWriter.java:[58,21] [HidingField] Hiding fields of superclasses may cause confusion and errors. This field is hiding a field of the same name in superclass: PartitionDataWriter` in https://github.com/apache/celeborn/pull/2555/commits/9f8b59fb8aaaf48c7b95b0af8c652497f0edd413. PTAL. cc @mridulm

SteNicholas commented:
@FMX,  I will introduce CI error-prone test in another pull request. Meanwhile, I rebased the main branch and fix `[WARNING] /Users/nicholas/Github/celeborn/worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/MapPartitionDataWriter.java:[58,21] [HidingField] Hiding fields of superclasses may cause confusion and errors. This field is hiding a field of the same name in superclass: PartitionDataWriter` in https://github.com/apache/celeborn/pull/2555/commits/1551d454624b56a8986294759d967764ef7f1812. PTAL. cc @mridulm

RexXiong reviewed (approved):
LGTM, Thanks!

mridulm reviewed (approved):
LGTM

mridulm commented:
Merged to main.
Thanks for working on this @SteNicholas !
Thanks for the reviews @cxzl25, @venkata91, @RexXiong , @FMX :-)
